nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp

```diff
@@ -70,12 +83,28 @@ class MotionModel
     // }
 
     const bool is_holo = isHolonomic();
+    float max_delta_vx = model_dt_ * control_constraints_.ax_max;
+    float min_delta_vx = model_dt_ * control_constraints_.ax_min;
+    float max_delta_vy = model_dt_ * control_constraints_.ay_max;
+    float min_delta_vy = model_dt_ * control_constraints_.ay_min;
+    float max_delta_wz = model_dt_ * control_constraints_.az;
     for (unsigned int i = 0; i != state.vx.shape(0); i++) {
       for (unsigned int j = 1; j != state.vx.shape(1); j++) {
-        state.vx(i, j) = state.cvx(i, j - 1);
-        state.wz(i, j) = state.cwz(i, j - 1);
+        float& vx_curr = state.vx(i, j - 1);
+        float& cvx_curr = state.cvx(i, j - 1);        
```

Comment:
I think you can store the `cvx_last` for `j - 1` so that you don't do redundant lookups 

If you have the loop doing: get last vx, get last cvx, clip, set vx. Then that clipped vx for that loop iteration is next loop's `j - 1`! That would reduce us back down to having no additional look ups which will probably fix the performance issue. 

You just need to make sure you set the initial values in the first for loop of the nested for look to set the conditions properly.